include/Gaffer/Animation.h

```diff
@@ -103,6 +103,37 @@ class GAFFER_API Animation : public ComputeNode
 
 				friend class CurvePlug;
 
+				typedef boost::intrusive::avl_set_member_hook<
+					boost::intrusive::link_mode<
+#					ifndef NDEBUG
+					boost::intrusive::safe_link
+#					else
+					boost::intrusive::normal_link
+#					endif
+					> > Hook;
+
+				// NOTE : this makes it possible to declare the avl set member hook as private
+				struct HookAccess
+				{
+					typedef Hook hook_type;
+					typedef Hook* hook_ptr;
+					typedef const Hook* const_hook_ptr;
+					typedef Key value_type;
+					typedef Key* pointer;
+					typedef const Key* const_pointer;
+
+					static hook_ptr to_hook_ptr( value_type& );
+					static const_hook_ptr to_hook_ptr( const value_type& );
+					static pointer to_value_ptr( hook_ptr );
+					static const_pointer to_value_ptr( const_hook_ptr );
+				};
```

Comment:
Had a little play with this to educate myself a bit more about `boost::intrusive`, and it seems like this might not actually be necessary, and we can use the regular `member_hook` directly because CurvePlug is a friend of Key? See this commit - https://github.com/johnhaddon/gaffer/commit/b57c157c1ea1528ecf783f3eaa91691a023d0294.

Reply:
done in latest commit

Reply:
Thanks - could you rebase quickly so that the container and undo commits are distinct? Currently this fix is in the undo commit.

One thing we do sometimes that seems to work pretty well is to use `git commit --fixup originalCommitSha` when addressing review comments. They're easy to see that way, and then `git rebase -i --autosquash` will automatically squash into the history we want.

Reply:
done